File: WebApi/Models/CustomerTypeDto.ts
export interface CustomerTypeGetDto {
  name: string;
  description?: string;
  createdDate: string;
  updatedDate?: string;
  createdByFullUser?: string;
  updatedByFullUser?: string;
  deletedByFullUser?: string;
}

export interface CustomerTypeCreateDto {
  name: string;
  description?: string;
}

export interface CustomerTypeUpdateDto {
  name: string;
  description?: string;
}

File: DTOs/CustomerDto/CustomerTypeDto.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace cms_webapi.DTOs
{
    public class CustomerTypeGetDto
    {
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
        
        // Full user information properties
        public string? CreatedByFullUser { get; set; }
        public string? UpdatedByFullUser { get; set; }
        public string? DeletedByFullUser { get; set; }
    }

    public class CustomerTypeCreateDto
    {
        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        [MaxLength(500)]
        public string? Description { get; set; }
    }

    public class CustomerTypeUpdateDto
    {
        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        [MaxLength(500)]
        public string? Description { get; set; }
    }
}
